osquery/remote/http_client.cpp

```diff
@@ -203,21 +203,27 @@ void Client::createConnection() {
 
 void Client::encryptConnection() {
   boost::asio::ssl::context ctx{boost::asio::ssl::context::sslv23};
+  const auto remote_hostname = *client_options_.remote_hostname_;
+
+  bool verify_hostname = false;
```

Comment:
This kinda makes me nervous, as it seems to suggest we don't verify the hostname by default. Under what circumstances would we _not_ want to verify? Wouldn't it just be with something like `--tls_allow_unsafe=true` or if the cert paths aren't set? Could we make this default to `true` and then only set to `false` if `client_options_.always_verify_peer_` is `false` or the like? What do you think?

Reply:
Well hostname is not verified until we set the verify_mode to `verify_peer` in the context, so I was just following that functionality.
Also the context is local to the encryptConnection function, so it cannot change outside.

That been said I could envision a different problem indeed, where there's a new case where verify_peer is enabled and the verify hostname enabling is forgotten.
I guess I could just always set it; would've preferred to have it clearly showing that it depends on the `verify_peer` mode, but the context cannot be queried it seems.

Reply:
Gotcha. I think I'd be in favor of always setting it, but that's because I'm having a lot of trouble imaging scenarios where you wouldn't want this verification enabled, and more over I think leveraging the `--tls_allow_unsafe=true` should be a sufficient work around for anyone wanting to do testing and the like?

Reply:
> Gotcha. I think I'd be in favor of always setting it, but that's because I'm having a lot of trouble imaging scenarios where you wouldn't want this verification enabled, and more over I think leveraging the `--tls_allow_unsafe=true` should be a sufficient work around for anyone wanting to do testing and the like?

Keep in mind that `verify_peer` already behaves like that in the context of the encryptConnection function: it's false by default and becomes true only if we explicitly set it.
I was reasoning completely in the context of that specific function because whatever external change is happening, it's not reflected in that function if not through the client_option settings.

I did not touch `verify_peer` because I was focusing on the hostname verification, but if we want to pursue this (correct) mentality, I'll change that to be enabled by default :).

Reply:
> it's false by default and becomes true only if we explicitly set it.

Ah ok, that makes sense.  Perhaps for the sake of this PR let's leave things as they are then so that we're pattern matching. For changing the default behavior of this to be enabled by default, maybe let's start a broader discussion in Slack to get more folks to weigh in, what do you think?

Reply:
Yeah makes sense!